src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Godot;
+
+/// <summary>
+///   Tweaked color picker defines some custom ColorPicker functions.
+/// </summary>
+public class TweakedColourPicker : ColorPicker
+{
+    private readonly CheckButton hsvCheckButton;
+
+    private readonly CheckButton rawCheckButton;
+
+    private readonly Button addPresetButton;
+
+    private bool hsvModeDisabled;
+
+    private bool rawModeDisabled;
+
+    public TweakedColourPicker()
+    {
+        var baseNode = GetChild(4).GetChild(4);
+        hsvCheckButton = baseNode.GetChild<CheckButton>(0);
+        hsvCheckButton.Connect("toggled", this, nameof(OnHSVButtonToggled));
+        rawCheckButton = baseNode.GetChild<CheckButton>(1);
+        rawCheckButton.Connect("toggled", this, nameof(OnRawButtonToggled));
+
+        addPresetButton = GetChild(7).GetChild<Button>(0);
+        addPresetButton.Connect("mouse_entered", this, nameof(OnMouseEnteredAddPresetButton));
+    }
+
+    [Export]
+    public bool HSVModeDisabled
+    {
+        get => hsvModeDisabled;
+        set
+        {
+            hsvModeDisabled = value;
+            UpdateControl();
+        }
+    }
+
+    [Export]
+    public bool RawModeDisabled
+    {
+        get => rawCheckButton.Disabled;
+        set
+        {
+            rawModeDisabled = value;
+            UpdateControl();
+        }
+    }
+
+    public new bool HsvMode
+    {
+        get => base.HsvMode;
+        set
+        {
+            if (hsvModeDisabled)
+                return;
+
+            base.HsvMode = value;
+        }
+    }
+
+    public new bool RawMode
+    {
+        get => base.RawMode;
+        set
+        {
+            if (rawModeDisabled)
+                return;
+
+            base.RawMode = value;
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+        UpdateControl();
+    }
+
+    public void OnMouseEnteredAddPresetButton()
+    {
+        addPresetButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_ADD_PRESET");
+    }
+
+    private void OnHSVButtonToggled(bool isOn)
+    {
+        if (!isOn && rawModeDisabled)
+            rawCheckButton.Disabled = true;
+    }
+
+    private void OnRawButtonToggled(bool isOn)
+    {
+        if (!isOn && hsvModeDisabled)
+            hsvCheckButton.Disabled = true;
+    }
+
+    private void UpdateControl()
+    {
+        hsvCheckButton.Disabled = hsvModeDisabled;
+        rawCheckButton.Disabled = rawModeDisabled;
```

Comment:
I think it's best/simpler if we just hide the buttons entirely since what we want is just so the player can't interface with the mode toggling. The color mode states is already handled in a certain way internally and I think it's a bit hacky if we force it to fit a custom behavior.

Reply:
If we hide it, the TextBox on the right will fill that place. (Too ugly!!!)

Reply:
A necessary trade-off for simplicity :) I actually won't mind that, or maybe you could put a spacer to keep the text box to the right in some way.

Reply:
Or alternatively why not let the buttons keep its toggle value while still being disabled? I think it makes more sense because what determines the modes is not necessarily the buttons themselves but rather some internal values in the godot logic. So basically, color modes can't be edited through GUI but is possible programmatically.

Reply:
Okay

Reply:
(How to add a placeholder?

Reply:
You mean like a translation string placeholder?

Reply:
I just don't know what can act as a spacer

Reply:
Ah, you could use a plain Control node for that, and set its size flag to `Fill` and `Expand` horizontally or vertically.

Reply:
Solved as 

> Or alternatively why not let the buttons keep their toggle value while still being disabled?